wpilib/wpilib/interruptablesensorbase.py

```diff
@@ -70,21 +67,17 @@ def allocateInterrupts(self, watcher):
             where the user program will have to explicitly wait for the interrupt
             to occur.
         """
-        if self.interrupt is not None:
-            raise ValueError("The interrupt has already been allocated")
 
         self.isSynchronousInterrupt = watcher
-        self._interrupt = hal.initializeInterrupts(watcher)
-        self._interrupt_finalizer = weakref.finalize(self, hal.cleanInterrupts,
```

Comment:
Is there a good reason to get rid of the finalizer?

Reply:
Only because it wasn't in upstream

Reply:
This would be where we should have comments saying 'this is python specific'... we should also document the finalizer pattern in the porting guide too I suppose.

The reason for having a finalizer is so that when the object is garbage collected, the hal resources will be cleaned up automatically.

@PeterJohnson you had added these originally, if the program exits without cleaning up hal resources, do they get cleaned up by the FPGA automatically, or does that cause issues?